src/Build/Construction/Solution/SolutionFile.cs

```diff
@@ -195,7 +200,63 @@ internal string FullPath
             {
                 // Should already be canonicalized to a full path
                 ErrorUtilities.VerifyThrowInternalRooted(value);
-                _solutionFile = value;
+                if (FileUtilities.IsSolutionFilterFilename(value))
+                {
+
+                    try
+                    {
+                        using JsonDocument text = JsonDocument.Parse(File.ReadAllText(value));
+                        JsonElement solution = text.RootElement.GetProperty("solution");
+                        _solutionFile = solution.GetProperty("path").GetString();
+                        if (!File.Exists(_solutionFile))
+                        {
+                            ProjectFileErrorUtilities.VerifyThrowInvalidProjectFile
+                            (
+                                false /* just throw the exception */,
```

Comment:
There is an overload of `ProjectFileErrorUtilities.ThrowInvalidProjectFile` with 5 parameters that is missing here. It is identical to what is written here but assumes false for the first parameter

Reply:
I see three `ProjectFileErrorUtilities.ThrowInvalidProjectFile`s, and they all take `bool condition` as the first argument.

Reply:
I think Dan's suggesting adding a new overload, and I agree.

Reply:
Ah, I see. It might be easiest to just make it a default argument.